src/main/webapp/user_script.js

```diff
@@ -0,0 +1,12 @@
+function loadProfile() {
+  fetch('/profile-data').then((response) => response.json()).then((profile) => {
+    const nameSection = document.getElementById('name');
+    nameSection.innerHTML = profile.name;
+
+    const usernameSection = document.getElementById('username');
+    usernameSection.innerHTML = profile.username;
+
+    const bioSection = document.getElementById('bio');
+    bioSection.innerHTML = profile.bio;
+  });
```

Comment:
add a catch after the fetch to print output if something failed, the code would look like this: 
```suggestion
  }).catch((error) => {
      console.log('Fetching profile data servlet failed: ' + error);
  });
```